homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,281 @@
+            if CONF_DEVICE_ID in user_input:
+                self._selected_device_event_code = user_input[CONF_DEVICE_ID]
+                self._selected_device = {}
+                return await self.async_step_set_device_options()
+
+            if not errors:
+                self.update_config_data(global_options=self._global_options)
+
+                return self.async_create_entry(title="", data={})
+
+        device_registry = await async_get_registry(self.hass)
+        device_entries = async_entries_for_config_entry(
+            device_registry, self._config_entry.entry_id
+        )
+        self._device_entries = device_entries
+
+        devices = {
+            entry.id: entry.name_by_user if entry.name_by_user else entry.name
+            for entry in device_entries
+        }
+
+        options = {
+            vol.Optional(CONF_DEBUG, default=self._config_entry.data[CONF_DEBUG]): bool,
+            vol.Optional(
+                CONF_AUTOMATIC_ADD, default=self._config_entry.data[CONF_AUTOMATIC_ADD],
+            ): bool,
+            vol.Optional(CONF_DEVICE_ID): str,
+            vol.Optional(CONF_DEVICE): vol.In(devices),
+            vol.Optional(CONF_REMOVE_DEVICE): vol.In(devices),
+        }
+
+        return self.async_show_form(
+            step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
+        )
+
+    async def async_step_set_device_options(self, user_input=None):
+        """Manage device options."""
+        errors = {}
+
+        if user_input is not None:
+            device_id = get_device_id(
+                self._rfxobj.device, data_bits=user_input.get(CONF_DATA_BITS)
+            )
+            try:
+                command_on = (
+                    int(user_input.get(CONF_COMMAND_ON), 16)
+                    if user_input.get(CONF_COMMAND_ON)
+                    else None
+                )
+                command_off = (
+                    int(user_input.get(CONF_COMMAND_OFF), 16)
+                    if user_input.get(CONF_COMMAND_OFF)
+                    else None
+                )
+            except (NameError, ValueError):
+                errors = {"base": "invalid_input_2262"}
+
+            if not errors:
+                devices = {}
+                devices[self._selected_device_event_code] = {
+                    CONF_DEVICE_ID: device_id,
+                    CONF_FIRE_EVENT: user_input.get(CONF_FIRE_EVENT, False),
+                    CONF_SIGNAL_REPETITIONS: user_input.get(CONF_SIGNAL_REPETITIONS, 1),
+                }
+
+                device = devices[self._selected_device_event_code]
+                if (
+                    user_input.get(CONF_OFF_DELAY_ENABLED)
+                    and user_input.get(CONF_OFF_DELAY) is not None
+                ):
+                    device[CONF_OFF_DELAY] = user_input[CONF_OFF_DELAY]
+                if user_input.get(CONF_DATA_BITS) is not None:
```

Comment:
What about empty values here? If you start typing something and erase it?

```suggestion
                if user_input.get(CONF_DATA_BITS):
```